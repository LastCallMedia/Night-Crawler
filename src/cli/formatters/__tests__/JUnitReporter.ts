import JUnitReporter from '../JUnitReporter';
import { file } from 'tmp-promise';
import { promises as fs } from 'fs';
import { TestResultMap, TestResult } from '../../../testing/TestContext';

function r(obj: { [k: string]: TestResult }): TestResultMap {
  return new Map(Object.entries(obj));
}

describe('JUnit Reporter', function() {
  const pass = r({
    pass: { pass: true }
  });
  const fail = r({
    fail: { pass: false, message: 'There was an error.' }
  });
  const mix = r({
    pass: { pass: true },
    fail: { pass: false, message: 'there was an error.' }
  });

  it('Should write a junit file containing results', async function() {
    const { path, cleanup } = await file();
    const reporter = new JUnitReporter(path);
    reporter.report('passing', pass);
    reporter.report('failing', fail);
    reporter.report('mixed', mix);
    await expect(reporter.stop()).resolves.toBeUndefined();
    await expect(fs.readFile(path, 'utf8')).resolves.toMatchSnapshot();
    await cleanup();
  });
});
