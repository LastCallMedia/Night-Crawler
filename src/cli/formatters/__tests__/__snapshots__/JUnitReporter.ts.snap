// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JUnit Reporter Should write a junit file containing results 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<testsuites>
  <testsuite name=\\"passing\\" tests=\\"1\\" failures=\\"0\\" errors=\\"0\\" skipped=\\"0\\" time=\\"0\\">
    <testcase classname=\\"pass\\" time=\\"0\\"/>
  </testsuite>
  <testsuite name=\\"failing\\" tests=\\"1\\" failures=\\"1\\" errors=\\"0\\" skipped=\\"0\\" time=\\"0\\">
    <testcase classname=\\"fail\\" time=\\"0\\">
      <failure/>
      <system-out><![CDATA[There was an error.]]></system-out>
    </testcase>
  </testsuite>
  <testsuite name=\\"mixed\\" tests=\\"2\\" failures=\\"1\\" errors=\\"0\\" skipped=\\"0\\" time=\\"0\\">
    <testcase classname=\\"pass\\" time=\\"0\\"/>
    <testcase classname=\\"fail\\" time=\\"0\\">
      <failure/>
      <system-out><![CDATA[there was an error.]]></system-out>
    </testcase>
  </testsuite>
</testsuites>"
`;
